sql-db/narayana-transactions/src/test/java/io/quarkus/ts/transactions/TransactionCommons.java

```diff
@@ -0,0 +1,208 @@
+    static final JaegerService jaeger = new JaegerService();
+
+    @Tag("QUARKUS-2492")
+    @Test
+    public void verifyNarayanaProgrammaticApproachTransaction() {
+        TransferDTO transferDTO = new TransferDTO();
+        transferDTO.setAccountFrom(ACCOUNT_NUMBER_MIGUEL);
+        transferDTO.setAccountTo(ACCOUNT_NUMBER_LOPE);
+        transferDTO.setAmount(100);
+
+        given()
+                .contentType(ContentType.JSON)
+                .body(transferDTO).post("/transfer/transaction")
+                .then().statusCode(HttpStatus.SC_CREATED);
+
+        AccountEntity miguelAccount = getAccount(ACCOUNT_NUMBER_MIGUEL);
+        Assertions.assertEquals(0, miguelAccount.getAmount(), "Unexpected account amount.");
+
+        AccountEntity lopeAccount = getAccount(ACCOUNT_NUMBER_LOPE);
+        Assertions.assertEquals(200, lopeAccount.getAmount(), "Unexpected account amount.");
+
+        JournalEntity miguelJournal = getLatestJournalRecord(ACCOUNT_NUMBER_MIGUEL);
+        Assertions.assertEquals(100, miguelJournal.getAmount(), "Unexpected journal amount.");
+    }
+
+    @Tag("QUARKUS-2492")
+    @Test
+    public void verifyNarayanaLambdaApproachTransaction() {
+        TransferDTO transferDTO = new TransferDTO();
+        transferDTO.setAccountFrom(ACCOUNT_NUMBER_GARCILASO);
+        transferDTO.setAccountTo(ACCOUNT_NUMBER_GARCILASO);
+        transferDTO.setAmount(100);
+
+        given()
+                .contentType(ContentType.JSON)
+                .body(transferDTO).post("/transfer/top-up")
+                .then().statusCode(HttpStatus.SC_CREATED);
+
+        AccountEntity garcilasoAccount = getAccount(ACCOUNT_NUMBER_GARCILASO);
+        Assertions.assertEquals(200, garcilasoAccount.getAmount(),
+                "Unexpected account amount. Expected 200 found " + garcilasoAccount.getAmount());
+
+        JournalEntity garcilasoJournal = getLatestJournalRecord(ACCOUNT_NUMBER_GARCILASO);
+        Assertions.assertEquals(100, garcilasoJournal.getAmount(), "Unexpected journal amount.");
+    }
+
+    @Tag("QUARKUS-2492")
+    @Test
+    public void verifyNarayanaRollbackWithProgrammaticApproach() {
+        TransferDTO transferDTO = new TransferDTO();
+        transferDTO.setAccountFrom(ACCOUNT_NUMBER_LUIS);
+        transferDTO.setAccountTo(ACCOUNT_NUMBER_LUIS);
+        transferDTO.setAmount(200);
+
+        given()
+                .contentType(ContentType.JSON)
+                .body(transferDTO).post("/transfer/withdrawal")
+                .then().statusCode(HttpStatus.SC_BAD_REQUEST);
+
+        AccountEntity fedorAccount = getAccount(ACCOUNT_NUMBER_LUIS);
+        Assertions.assertEquals(100, fedorAccount.getAmount(), "Unexpected account amount.");
+
+        given().get("/transfer/journal/latest/" + ACCOUNT_NUMBER_LUIS)
+                .then()
+                .statusCode(HttpStatus.SC_NO_CONTENT);
+    }
+
+    @Tag("QUARKUS-2492")
+    @Test
+    public void smokeTestNarayanaProgrammaticTransactionTrace() {
```

Comment:
```suggestion
    public void tracing() {
```

Reply:
same as metrics. Here with the name what we are saying is, what we are having a look at the middleware. This is a tracing smoke test of Narayana transactions, of the programmatic API.